zorg/buildbot/builders/UnifiedTreeBuilder.py

```diff
@@ -567,3 +571,448 @@ def getCmakeWithNinjaMultistageBuildFactory(
+            flunkOnFailure  = False,
+            doStepIf        = lambda step, clean = clean: clean or step.getProperty("clean_obj") == True
+        ),
+    ])
+
+    # Get the source code steps at first. We share it between all stages.
+
+    # Add the Git step.
+    extra_git_args = extra_git_args or {}
+
+    f.addGetSourcecodeSteps(**extra_git_args)
+
+    # Add the custom pre-configuration workflow extension steps, if specified.
+    if pre_configure_steps:
+        extend_with_custom_steps(f, pre_configure_steps)
+
+    # Configure MSVC environment at first if requested.
+    if vs:
+        f.addStep(
+            steps.SetPropertyFromCommand(
+                name            = "set-pros.vs_env",
+                command         = builders_util.getVisualStudioEnvironment(vs, vs_arch),
+                extract_fn      = builders_util.extractVSEnvironment,
+                env             = env
+            ))
+        env = util.Property('vs_env')
+
+    # Build the CMake command definitions.
+    cmake_definitions = cmake_definitions or dict()
+    assert isinstance(cmake_definitions, dict), "The CMake definitions argument must be a dictionary."
+    cmake_options = cmake_options or list()
+    assert isinstance(cmake_options, list), "The CMake options argument must be a list."
+
+    if not "LLVM_ENABLE_PROJECTS" in cmake_definitions and f.enable_projects:
+        cmake_definitions.update({ "LLVM_ENABLE_PROJECTS" : ";".join(f.enable_projects) })
+
+    if not "LLVM_ENABLE_RUNTIMES" in cmake_definitions and f.enable_runtimes:
+        cmake_definitions.update({ "LLVM_ENABLE_RUNTIMES" : ";".join(f.enable_runtimes) })
+
+    if not "CMAKE_INSTALL_PREFIX" in cmake_definitions and f.install_dir:
+        cmake_definitions.update({ "CMAKE_INSTALL_PREFIX" : LLVMBuildFactory.pathRelativeTo(
+                                                                f.install_dir,
+                                                                f.obj_dir) })
+        # Remove install directory.
+        f.addSteps([
+            steps.RemoveDirectory(
+                name            = f"clean-install-dir",
+                dir             = util.Interpolate(f.install_dir),
+                description     = ["Remove", util.Interpolate(f.install_dir), "directory"],
+                haltOnFailure   = False,
+                flunkOnFailure  = False,
+                doStepIf        = lambda step, clean = clean: clean or step.getProperty("clean_obj") == True
+            ),
+        ])
+
+    # Set proper defaults.
+    if allow_cmake_defaults:
+        if not "CMAKE_BUILD_TYPE" in cmake_definitions:
+            cmake_definitions.update({ "CMAKE_BUILD_TYPE" : "Release" })
+        if not "LLVM_ENABLE_ASSERTIONS" in cmake_definitions:
+            cmake_definitions.update({ "LLVM_ENABLE_ASSERTIONS" : "ON" })
+        if not "LLVM_LIT_ARGS" in cmake_definitions and checks:
+            cmake_definitions.update({ "LLVM_LIT_ARGS" : "-v -vv" })
```

Comment:
`-v` implies `-vv`. Specifying the both is redundant. Nice to fix but is not a blocker for this patch, though.

Reply:
Updated accordingly